### PR TITLE
Use env var to disable loading dotenv

### PR DESCRIPTION
**Pull request by lepture**
Fix issue: https://github.com/pallets/flask/issues/2722

### CODE REVIEW

**Comment by lepture** [comment]
I need to add a test case.

File: flask/cli.py [hunk 1]
```diff
@@ -544,7 +544,10 @@ def main(self, *args, **kwargs):
         # script that is loaded here also attempts to start a server.
         os.environ['FLASK_RUN_FROM_CLI'] = 'true'
 
-        if self.load_dotenv:
+        val = os.environ.get('FLASK_DONT_LOAD_ENV')
+        load_dotenv =  not val or val in ('0', 'false', 'no')
```
Comment (ThiefMaster) [inline flask/cli.py]: too much space. also, is this the same logic as pipenv uses?

Comment (lepture) [reply flask/cli.py]: @ThiefMaster No, this is the same logic `FLASK_DEBUG` is using.

Comment (ThiefMaster) [reply flask/cli.py]: ok fine for me as well

**Comment by lepture** [comment]
Oops. Tests should not fail since I didn't add any changes to break the test. Weird.

**Comment by lepture** [comment]
@davidism any idea why tests failed? I've tested master branch on my local machine, tests failed too.

**Comment by davidism** [comment]
I can take a look tomorrow. The most common issue I had with CLI tests was that a test would forget to reset the env vars. Not sure if that's related here.

**Comment by lepture** [comment]
The weird thing is master branch's tests fail on my local machine.

File: flask/cli.py (hunk 1 above)
Comment (davidism) [inline flask/cli.py]: `load_dotenv` is the name of the function that will be called later, it's been overwritten as a boolean.

**Comment by davidism** [comment]
Needs a test.

**Comment by rhymes** [comment]
Shouldn't it be called "FLASK_IGNORE_DOTENV" or "FLASK_SKIP_DOTENV" ?

"Flask don't load env" it's not exactly accurate because the flag purpose is to skip reading variables from .env, not to skip the shell environment.

**Comment by davidism** [comment]
Discussion at #2722, it was chosen to match pipenv's `PIPENV_DONT_LOAD_ENV`. I agree that either of those sound better though; I like `SKIP`.

**Comment by davidism** [comment]
Added the check to `app.run` as well in 6e1e3e03ca188a4d9b3aef493e572157c8ae2313.

**Pull request merged**